tests/kafkatest/services/trogdor/kibosh.py

```diff
@@ -133,9 +133,12 @@ def set_faults(self, node, specs):
         :param node:        The node.
         :param spec:        An array of FaultSpec objects describing the faults.
         """
-        fault_array = [spec.kibosh_message for spec in specs]
-        obj = { 'faults': fault_array }
-        obj_json = json.dumps(obj)
+        if len(specs) == 0:
+            obj_json = "{}"
```

Comment:
What is the behavior when JSON is sent with other properties set, but not "faults". Will this also clear the faults array?

Reply:
yes, this would clear the faults array.  Kibosh basically stores the bytes that are sent (so it doesn't change them).